drivers/SmartThings/zigbee-fan/src/kichler-fan-light/init.lua

```diff
@@ -0,0 +1,187 @@
+    device:set_field('LAST_DIM_LEVEL', trim_level, {persist = true})
+  elseif command.args.level < 1 then
+    local level = 0
+    device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, level, command.args.rate or 0xFFFF))
+  else
+    local level = math.floor((command.args.level/100.0) * 254 )
+    device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, level, command.args.rate or 0xFFFF))
+    device:set_field('LAST_DIM_LEVEL', command.args.level, {persist = true})
+  end
+end
+
+local function fan_speed_handler(driver, device, command)
+  if command.args.speed < 4 then
+    device:send(FanControl.attributes.FanMode:write(device, command.args.speed))
+    device:send(FanControl.attributes.FanMode:read(device))
+  end
+end
+
+-- ZIGBEE HANDLERS
+local function zb_fan_control_handler(driver, device, value, zb_rx)
+  if value.value < 4 then
+    device:emit_event(capabilities.fanSpeed.fanSpeed(value.value))
+    local evt = capabilities.switch.switch(value.value > 0 and 'on' or 'off', { visibility = { displayed = true } })
+    device:emit_component_event(device.profile.components.main, evt)
+    device:emit_component_event(device.profile.components.main, capabilities.fanSpeed.fanSpeed(value.value))
+    if value.value > 0 then
+      device:set_field('LAST_FAN_SPD', value.value, {persist = true})
+    end
+  end
+end
+
+local function zb_level_handler(driver, device, value, zb_rx)
+  local evt = capabilities.switchLevel.level(math.floor(0.5 + (value.value / 254.0) * 100))
+  device:emit_component_event(device.profile.components.light, evt)
+end
+
+local function zb_onoff_handler(driver, device, value, zb_rx)
+  local attr = capabilities.switch.switch
+  local evt = value.value and attr.on() or attr.off()
+  device:emit_component_event(device.profile.components.light, evt)
+end
+
+local function info_changed(driver, device, event, args)
+  if device.preferences ~= nil then
+    local current_level = device:get_latest_state('light', capabilities.switchLevel.ID, capabilities.switchLevel.level.NAME)
+    local new_trim = tonumber(device.preferences.trim)
+    local old_trim = tonumber(args.old_st_store.preferences.trim)
+    if args.old_st_store.preferences.trim ~= device.preferences.trim then
+      if old_trim < new_trim then
+        if new_trim >= current_level then
+          local newlevel = math.floor((new_trim/100.0) * 254 )
+          device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, newlevel, 0))
+        else
+          local newlevel = math.floor((current_level/100.0) * 254 )
+          device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, newlevel, 0))
+        end
+      else
+        local newlevel = math.floor((current_level/100.0) * 254 )
+        device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, newlevel, 0))
+      end
```

Comment:
```suggestion
      local newlevel = math.floor((current_level/100.0) * 254 )
      if old_trim < new_trim then
        if new_trim >= current_level then
          local newlevel = math.floor((new_trim/100.0) * 254 )
        end
      device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, newlevel, 0))
```

Reply:
I modified it with the code you suggested